storage/replica.go

```diff
@@ -1299,12 +1299,20 @@ func (r *Replica) maybeGossipSystemConfig() {
 	r.maybeGossipSystemConfigLocked()
 }
 
+// IsLeader returns true if the node has a leader lease on the replica.
+func (r *Replica) IsLeader() bool {
```

Comment:
@tschottdorf can you review this part. All the tests work, but I want to make sure this is the right function to call to check for replica leader.